cmd/egress/cmd.go

```diff
@@ -63,20 +68,68 @@ are set correctly before execution.
 				fmt.Printf("Unable to build logger: %s\n", err.Error())
 				os.Exit(1)
 			}
-			logger.Info(ctx, "Using region: %s", config.region)
+
 			var creds interface{}
-			if config.awsProfile != "" {
-				creds = config.awsProfile
-				logger.Info(ctx, "Using AWS profile: %s", config.awsProfile)
+
+			if !config.gcp {
+				//AWS stuff
+				if config.awsProfile != "" {
+					creds = config.awsProfile
+					logger.Info(ctx, "Using AWS profile: %s", config.awsProfile)
+				} else {
+					creds = credentials.NewStaticCredentialsProvider(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
+				}
+				if err != nil {
+					logger.Error(ctx, err.Error())
+					os.Exit(1)
+				}
+
 			} else {
-				creds = credentials.NewStaticCredentialsProvider(os.Getenv("AWS_ACCESS_KEY_ID"), os.Getenv("AWS_SECRET_ACCESS_KEY"), os.Getenv("AWS_SESSION_TOKEN"))
+				// GCP stuff
+
+				if os.Getenv("GCP_VPC_NAME") == "" {
+					logger.Error(ctx, "please set environment variable GCP_VPC_NAME to the name of VPC")
+					os.Exit(1)
+				}
+
+				if os.Getenv("GCP_PROJECT_ID") == "" {
+					logger.Error(ctx, "please set environment variable GCP_PROJECT_ID to the project ID of VPC")
+					os.Exit(1)
+				}
+				creds = &google.Credentials{ProjectID: os.Getenv("GCP_PROJECT_ID")}
+
+				if os.Getenv("GOOGLE_APPLICATION_CREDENTIALS") == "" {
+					logger.Info(ctx, "GOOGLE_APPLICATION_CREDENTIALS not set; using service account attached to %s", os.Getenv("GCP_PROJECT_ID"))
+				} else {
+					logger.Info(ctx, "Using GCP credential json file from %s", os.Getenv("GOOGLE_APPLICATION_CREDENTIALS"))
+				}
+
+				//gcp region
+				val, present := os.LookupEnv(GcpRegionEnvVarStr)
+				if present {
+					config.region = val
+				} else if config.region == regionDefault {
+					config.region = GcpRegionDefault
+				}
+
+				//default gcp machine e2
```

Comment:
Would be cleaner to not use the default the option and set it to the proper value in case no value is provided (this would allow to not override anything in case the "t3.micro" value is passewd for GCP on purpose)

Reply:
Now default values are set in config type (aws/gcp)